src/sync/order_rewards.py

```diff
@@ -26,37 +26,31 @@ class OrderbookDataHandler(RecordHandler):  # pylint:disable=too-few-public-meth
     """
 
     def __init__(
-        self, block_range: BlockRange, config: SyncConfig, order_rewards: DataFrame
+        self,
+        file_manager: FileIO,
+        block_range: BlockRange,
+        config: SyncConfig,
+        order_rewards: DataFrame,
     ):
         super().__init__(block_range, SYNC_TABLE, config)
         log.info(f"Handling {len(order_rewards)} new records")
-        self.order_rewards = order_rewards
+        self.file_manager = file_manager
+        self.order_rewards = OrderRewards.from_pdf_to_dune_records(order_rewards)
 
     def num_records(self) -> int:
         return len(self.order_rewards)
 
     def write_found_content(self) -> None:
-        path = self.file_path
-        if not os.path.exists(path):
-            log.info(f"creating write path {path}")
-            os.makedirs(path)
-
-        self.order_rewards.to_json(
-            os.path.join(self.file_path, self.content_filename),
-            orient="records",
-            lines=True,
+        self.file_manager.write_ndjson(
+            data=self.order_rewards, name=self.content_filename
         )
 
     def write_sync_data(self) -> None:
         # Only write these if upload was successful.
-        config = self.config
-        column = config.sync_column
-        with open(
-            os.path.join(self.file_path, config.sync_file), "w", encoding="utf-8"
-        ) as sync_file:
-            writer = csv.DictWriter(sync_file, fieldnames=[column], lineterminator="\n")
-            writer.writeheader()
-            writer.writerows([{column: str(self.block_range.block_to)}])
```

Comment:
One bonus of this change is that this method is now identical to the one for app_data and can be moved into the "super class" (RecordHandler). Unfortunately, the same is not true for `write_found_content`.